src/Compiler/Facilities/DiagnosticsLogger.fs

```diff
@@ -877,14 +877,23 @@ type StackGuard(maxDepth: int, name: string) =
 
     static member val DefaultDepth =
 #if DEBUG
-        GetEnvInteger "FSHARP_DefaultStackGuardDepth" 50
+        GetEnvInteger "FSHARP_DefaultStackGuardDepth" 55
 #else
-        GetEnvInteger "FSHARP_DefaultStackGuardDepth" 100
+        let defaultStackGuardDepth = StackGuard.GetOsDependentDepth(100, 110, 110, 100)
+        GetEnvInteger "FSHARP_DefaultStackGuardDepth" defaultStackGuardDepth
 #endif
 
     static member GetDepthOption(name: string) =
         GetEnvInteger ("FSHARP_" + name + "StackGuardDepth") StackGuard.DefaultDepth
 
+    static member GetOsDependentDepth(mac: int, unix: int, win: int, other: int) : int =
```

Comment:
This should be inline, also should be taking an environment variable as parameter and taking value from env inside.

Reply:
Cool, yeah, looks nice. Just the inlining is an issue:

```
C:\Users\schae\src\fsharp\src\Compiler\Facilities\DiagnosticsLogger.fs(882,9): warning FS1116: A value marked as 'inline' has an unexpected value [C:\Users\schae\src\fsharp\src\Compiler\FSharp.Compiler.Service.fsproj::TargetFramework=netstandard2.0]
C:\Users\schae\src\fsharp\src\Compiler\Facilities\DiagnosticsLogger.fs(882,9): error FS1114: The value 'FSharp.Compiler.DiagnosticsLogger.StackGuard.GetOsDependentDepth' was marked inline but was not bound in the optimization environment [C:\Users\schae\src\fsharp\src\Compiler\FSharp.Compiler.Service.fsproj::TargetFramework=netstandard2.0]
C:\Users\schae\src\fsharp\src\Compiler\Facilities\DiagnosticsLogger.fs(888,26): error FS1113: The value 'GetOsDependentDepth' was marked inline but its implementation makes use of an internal or private function which is not sufficiently accessible [C:\Users\schae\src\fsharp\src\Compiler\FSharp.Compiler.Service.fsproj::TargetFramework=netstandard2.0]  
C:\Users\schae\src\fsharp\src\Compiler\Facilities\DiagnosticsLogger.fs(882,9): error FS1118: Failed to inline the value 'GetOsDependentDepth' marked 'inline', perhaps because a recursive value was marked 'inline' [C:\Users\schae\src\fsharp\src\Compiler\FSharp.Compiler.Service.fsproj::TargetFramework=netstandard2.0]
```

Reply:
Hm, that is awkward...It's fine then, JIT will be able to inline it likely.

Reply:
I would prefer not to have code that compiles on windows but not the mac.  So probably keep the limits the same.

Reply:
This is all the way around, on Mac the limit is lower -> we will spawn another thread earlier. So it will work, but will have different frames limit on different OSs